cmd/scollector/collectors/vsphere.go

```diff
@@ -61,6 +66,82 @@ type DatastoreHostMount struct {
 	} `xml:"mountInfo"`
 }
 
+type HostScsiDiskPartition struct {
+	DiskName  string `xml:"diskName"`
+	Partition int    `xml:"partition"`
+}
+
+type HostVmfsVolume struct {
+	Type   string                  `xml:"type"`
+	Name   string                  `xml:"name"`
+	Uuid   string                  `xml:"uuid"`
+	Extent []HostScsiDiskPartition `xml:"extent"`
+}
+
+type VmfsDatastoreInfo struct {
+	Name string          `xml:"name"`
+	Url  string          `xml:"url"`
+	Vmfs *HostVmfsVolume `xml:"vmfs,omitempty"`
+}
+
+func vsphereStores(v *vsphere.Vsphere, storeKey map[string]string) error {
+	res, err := v.Info("Datastore", []string{
+		"name",
+	})
+	if err != nil {
+		return err
+	}
+	for _, r := range res {
+		var name string
+		for _, p := range r.Props {
+			if p.Name == "name" {
+				name, _ = opentsdb.Clean(p.Val.Inner)
+				break
+			}
+		}
+		if name == "" {
+			continue
+		}
+		storeKey[r.ID] = name
+	}
+
+	res, err = v.Datastores(storeKey)
+	if err != nil {
+		return err
+	}
+	for _, r := range res {
+		var name string
+		for _, p := range r.Props {
+			if p.Name == "name" {
+				name, _ = opentsdb.Clean(p.Val.Inner)
+				break
+			}
+		}
+		if name == "" {
+			continue
+		}
+		storeKey[r.ID] = name
+		for _, p := range r.Props {
+			switch p.Val.Type {
```

Comment:
Had this added for future HostNasVolume and HostLocalFileSystemVolume support, but we don't use those and also there is nothing interesting there, so just deleted it. Will change it, thanks.